main.py

```diff
@@ -1,30 +1,91 @@
 """
 This is an Add-On for DocumentCloud which wraps the tabula-py library
 https://github.com/chezou/tabula-py
-
-It attempts to convert tables from PDFs to CSVs
+It allows you to export tables from a set of PDFs into CSVs and
+download the resulting CSVs as a zip file.
 """
 
+import os
+import sys
+import fnmatch
+import zipfile
+import subprocess
+from urllib.parse import urlparse
+import requests
 import tabula
 from documentcloud.addon import AddOn
-
+from clouddl import grab
 
 class Tabula(AddOn):
-    """A tabula Add-On for DocumentCloud"""
-
-    def main(self):
-
-        document = self.client.documents.get(self.documents[0])
-        with open(f"{document.slug}.pdf", "wb") as pdf_file:
-            pdf_file.write(document.pdf)
+    """A tabula PDF table extraction Add-On for DocumentCloud"""
 
-        tabula.convert_into(
-            f"{document.slug}.pdf", f"{document.slug}.csv", output_format="csv", pages="all"
-        )
+    def fetch_template(self, url):
+        """Fetch the template from either Dropbox, Google Drive, or any public URL"""
+        os.makedirs(os.path.dirname("./out/"), exist_ok=True)
+        if(grab(url, "./out/")):
+            for file in os.listdir('./out/'):
+                if fnmatch.fnmatch(file, '*.json'):
+                   template_path = os.path.join('./out/', file)
+                   os.rename(template_path, os.path.join('./out/', 'template.json'))
+                   return True
```

Comment:
You need to handle the case where they use an upload link but there is no `*.json`  file - may want to set an informative message to let them know that no json file was found.  They may be confused if they supply a URL but it silently falls back to not using it.